### PR TITLE
Wb store id option

### PR DESCRIPTION
**Pull request by wbobillet**

### CODE REVIEW

File: alica_engine/src/engine/teammanager/TeamManager.cpp [hunk 1]
```diff
@@ -84,12 +84,24 @@ void TeamManager::readSelfFromConfig()
 {
     essentials::SystemConfig& sc = essentials::SystemConfig::getInstance();
     const std::string localAgentName = _engine->getRobotName();
-    int id = sc["Local"]->tryGet<int>(-1, "Local", "ID", NULL);
-    if (id != -1) {
+    
+    constexpr auto notAValidID = (unsigned long long)(-1);
```
Comment (hendrik-skubch) [inline alica_engine/src/engine/teammanager/TeamManager.cpp]: uint64_t　is the cleaner choice imo.
also c-style casts?
I'd recommend clang-tidy 
/ramble off ;)

File: alica_engine/src/engine/teammanager/TeamManager.cpp [hunk 2]
```diff
@@ -84,12 +84,24 @@ void TeamManager::readSelfFromConfig()
...
+    unsigned long long id = sc["Local"]->tryGet<unsigned long long>(notAValidID, "Local", "ID", NULL);
+    if (id != notAValidID) {
         _localAnnouncement.senderID = _engine->getId(id);
     } else {
         _localAnnouncement.senderID = _engine->generateId(DEFAULT_AGENT_ID_SIZE);
         ALICA_DEBUG_MSG("tm: Auto generated id " << _localAnnouncement.senderID);
+        bool persist_id = sc["Alica"]->tryGet<bool>(false, "Alica", "PersistID", NULL);
```
Comment (abhi1287) [inline alica_engine/src/engine/teammanager/TeamManager.cpp]: persistId by naming convention.

File: alica_engine/src/engine/teammanager/TeamManager.cpp (hunk 1 above)
Comment (abhi1287) [inline alica_engine/src/engine/teammanager/TeamManager.cpp]: Can use ULLONG_MAX

Comment (hendrik-skubch) [reply alica_engine/src/engine/teammanager/TeamManager.cpp]: better yet std::numeric_limits<uint64_t>::max();

**Review by kota-rr — APPROVED** [review]

**Review by abhi1287 — APPROVED** [review]

**Pull request merged**